File: Front/src/api/endpoints/forecast.ts
// AI 수요예측·발주추천 API — api_spec.md §8 (BE 프록시, /api/forecast/*).
import { api } from "../../lib/api";

export interface TopFactor {
  feature: string;
  label: string;
  pct: number;
}

export interface Explanation {
  baseline: string;
  deviation_vs_baseline: number;
  top_factors: TopFactor[];
  sentence: string;
}

export interface DailyPrediction {
  target_date: string;
  horizon_days: number;
  predicted_sales: number;
  interval_p10: number;
  interval_p90: number;
  is_low_confidence: boolean;
  low_confidence_reason: string | null;
  explanation: Explanation;
}

export interface ForecastPredictResponse {
  predictions: DailyPrediction[];
}

export async function getForecastPredict(): Promise<ForecastPredictResponse> {
  return api.get("forecast/predict").json<ForecastPredictResponse>();
}

export interface MenuForecastItem {
  menu_id: string;
  menu_name: string;
  expected_quantity: number;
}

export interface OrderRecommendation {
  item_id: string;
  item_name: string;
  unit: string;
  recommended_quantity: number;
  expected_stockout_date: string | null;
  lead_time_days: number;
  safety_stock: number;
  config_status: string;
  recommendation_reason: string;
}

export interface AIRecommendResponse {
  target_dates: string[];
  is_low_confidence: boolean;
  low_confidence_reason: string | null;
  menu_forecast: MenuForecastItem[];
  recommendations: OrderRecommendation[];
}

export async function getAIRecommend(): Promise<AIRecommendResponse> {
  return api.get("forecast/recommend").json<AIRecommendResponse>();
}

File: Back/app/schemas/forecast.py
"""AI 수요예측·발주추천 DTOs — api_spec.md §8 계약을 BE 응답 형태로 재노출."""
from __future__ import annotations

from datetime import date

from pydantic import BaseModel


class TopFactor(BaseModel):
    feature: str
    label: str
    pct: float


class Explanation(BaseModel):
    baseline: str
    deviation_vs_baseline: float
    top_factors: list[TopFactor]
    sentence: str


class DailyPrediction(BaseModel):
    target_date: date
    horizon_days: int
    predicted_sales: int
    interval_p10: int
    interval_p90: int
    is_low_confidence: bool
    low_confidence_reason: str | None = None
    explanation: Explanation


class ForecastPredictResponse(BaseModel):
    predictions: list[DailyPrediction]


class MenuForecastItem(BaseModel):
    menu_id: str
    menu_name: str
    expected_quantity: float


class OrderRecommendation(BaseModel):
    item_id: str
    item_name: str
    unit: str
    recommended_quantity: float
    expected_stockout_date: date | None
    lead_time_days: int
    safety_stock: float
    config_status: str
    recommendation_reason: str


class AIRecommendResponse(BaseModel):
    target_dates: list[date]
    is_low_confidence: bool
    low_confidence_reason: str | None = None
    menu_forecast: list[MenuForecastItem]
    recommendations: list[OrderRecommendation]
